internal/controller/flow_cilium.go

```diff
@@ -187,11 +187,7 @@ func (fm *CiliumFlowCollector) exportCiliumFlows(ctx context.Context, sm *stream
 		if ciliumFlow == nil {
 			continue
 		}
-		err = sm.sendNetworkFlowRequest(fm.logger, ciliumFlow)
-		if err != nil {
-			fm.logger.Error("Cannot send cilium flow", zap.Error(err))
-			return err
-		}
+		sm.flowChannel <- ciliumFlow
```

Comment:
Never do a blocking write into a channel ignoring `ctx`.
Just wrap that into a method of `FlowCache` as I suggested above.
```suggestion
		sm.flowCache.CacheFlow(ctx, ciliumFlow)
```